### ISSUE TITLE
Request.clone() not duplicating body stream

### ISSUE THREAD

**Issue by sergiga** [denoland/deno#5951]
I'm trying to make a proxy server that analyzes the response body for each request.

The problem comes when using `res.clone()` after the request has been proxied, I try to save the body data after the server responds, and then return the response to the client like this

```
const body = await res.clone().body.json().catch(console.error)
await req.respond(res.clone()).catch(console.error)
```

And the server throws `error: Uncaught TypeError: Failed to execute 'clone' on 'Response': Response body is already used` because the body stream seems to be the same and it has finished reading.
I've tried `getReader()` too but it's not implemented yet

Here is a minimum reproducible [gist](https://gist.github.com/sergiga/eea389f13147c6841fe726514d390a16)

Meanwhile... is there any workaround of this issue?

Thanks!

**Comment by marcosc90** [comment]
This is fixed by https://github.com/denoland/deno/pull/5787, will be included in `1.0.3`

```
async function proxyRequest(req) {
	const url = `${proxyServerAddress}${req.url}`;
	console.log(url);
	const res = await fetch(url, {
		method: req.method,
		// headers: req.headers,
	});
	const body = await res.clone().json().catch(console.error);

	const originalBody = new Uint8Array(await res.arrayBuffer());
	await req.respond({ body: originalBody }).catch(console.error);
}
```

And if https://github.com/denoland/deno/pull/5789 gets merged you'll be able to do the following: 

```
await req.respond({ body: Deno.toReader(res.body) }).catch(console.error);
```
Later we may improve `Response.respond`'s `body` to take a `ReadableStream`

**Comment by bartlomieju** [comment]
Fixed in #5787
